### DISCUSSION TITLE
pandoc.utils.stringify() behaviour change in pandoc 3.6 (probably correct!)

### DISCUSSION THREAD

**Discussion by wlupton** [jgm/pandoc#10447] [General]
I have noticed that [`pandoc.utils.stringify()`](https://pandoc.org/lua-filters.html#pandoc.utils.stringify) can behave differently in pandoc 3.6. Probably it now behaves as documented, and previously was too lax?

The case that I hit was an attempt to stringify a [`Caption`](https://pandoc.org/lua-filters.html#type-caption). Previously this worked but I guess it shouldn't, because you're only supposed to be able to stringify AST elements, and captions are objects with `long` (`Blocks`) and `short` (`Inlines`) fields.

However, should the behaviour have changed? I'm not sure.

---

Here are my tests.

```
% cat figure.md
![Cat](Cat.png)
```

```
% cat caption.lua
function Figure(figure)
    local caption = pandoc.utils.stringify(figure.caption)
    io.stderr:write(string.format('#### caption = %s\n', caption))
end
```

```
% pandoc-3.5 figure.md -L caption.lua -t html -o /dev/null
#### caption = Cat
```

```
% pandoc-3.6 figure.md -L caption.lua -t html -o /dev/null
Error running filter caption.lua:
table expected, got Caption
stack traceback:
	caption.lua:2: in function 'Figure'

**Comment by jgm** [comment]
I defer to @tarleb on this.
I don't think there was a change on the Haskell side, but maybe one on the Lua side.

**Reply by tarleb** [reply]
Captions used to be represented as Lua tables with two keys, `long` and `short` (the latter can be missing). The `stringify` function interpreted that as metadata and was able to stringify it that way. Now that captions are represented as userdata values, this no longer works. We now need a special case for that.

I'll push a fix shortly.
